pandas/core/indexes/base.py

```diff
@@ -472,7 +472,7 @@ def __new__(
                         pass
 
                     return Index(subarr, copy=copy, dtype=object, name=name)
-                elif inferred in ["floating", "mixed-integer-float"]:
+                elif inferred in ["floating", "mixed-integer-float", "integer-na"]:
```

Comment:
ccan you add a TODO that we can return a Integer Array at some point

Reply:
TODO added